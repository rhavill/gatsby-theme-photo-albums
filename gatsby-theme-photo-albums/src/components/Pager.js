import React from 'react'
import PropTypes from 'prop-types'
import {Link} from 'gatsby'
import PreviousPageText from './PreviousPageText'
import NextPageText from './NextPageText'

const Pager = ({path, currentPage, numPages}) => {
  const pathWithoutPageNumber = path.replace(/\/?\d+$/, '')
  const isFirst = currentPage === 1
  const isLast = currentPage === numPages
  const prevPage = currentPage - 1 === 1 
    ? pathWithoutPageNumber 
    : pathWithoutPageNumber + '/' + (currentPage - 1).toString()
  const nextPage = pathWithoutPageNumber + '/' + (currentPage + 1).toString()

  return (
    <div className='pager'>
      {!isFirst && (
        <Link to={prevPage} rel='prev'>
          <PreviousPageText />
        </Link>
      )}
      {!isLast && (
        <Link to={nextPage} rel='next'>
          <NextPageText />
        </Link>
      )}
    </div>
  )
}

Pager.propTypes = {
  path: PropTypes.string.isRequired,
  currentPage: PropTypes.number.isRequired,
  numPages: PropTypes.number.isRequired,
}

export default Pager