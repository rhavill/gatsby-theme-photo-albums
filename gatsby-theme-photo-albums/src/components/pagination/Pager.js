import React from 'react'
import PropTypes from 'prop-types'
import {Link} from 'gatsby'
import PagerStyles from '../../styles/Pager'
import PreviousPageText from './PreviousPageText'
import NextPageText from './NextPageText'
import {getPagerUrls} from '../../util/url-text'

const Pager = ({path, currentPage, numPages}) => {
  const {prev, next} = getPagerUrls(path, currentPage, numPages)

  return (
    <PagerStyles>
      {prev ? (
        <Link to={prev} rel='prev'>
          <PreviousPageText />
        </Link>
      ) : null}
      {next ? (
        <Link to={next} rel='next'>
          <NextPageText />
        </Link>
      ) : null}
    </PagerStyles>
  )
}

Pager.propTypes = {
  path: PropTypes.string.isRequired,
  currentPage: PropTypes.number.isRequired,
  numPages: PropTypes.number.isRequired,
}

export default Pager