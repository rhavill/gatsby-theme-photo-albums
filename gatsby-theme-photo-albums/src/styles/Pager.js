/** @jsx jsx */
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'

const Pager = ({children}) => 
  <div sx={{variant: 'layout.pager'}}>
    {children}
  </div>

Pager.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default Pager