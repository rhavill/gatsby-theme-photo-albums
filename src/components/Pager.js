import React from 'react'
import PropTypes from 'prop-types'
import {Link} from 'gatsby'

const Pager = ({path, currentPage, numPages}) => {
  const pathWithoutPageNumber = path.replace(/\/?\d+$/, '')
  const isFirst = currentPage === 1
  const isLast = currentPage === numPages
  const prevPage = currentPage - 1 === 1 
    ? pathWithoutPageNumber 
    : pathWithoutPageNumber + '/' + (currentPage - 1).toString()
  const nextPage = pathWithoutPageNumber + '/' + (currentPage + 1).toString()

  return (
    <div className='pager'>
      {!isFirst && (
        <Link to={prevPage} rel='prev'>
          ← Previous Page
        </Link>
      )}
      {!isLast && (
        <Link to={nextPage} rel='next'>
          Next Page →
        </Link>
      )}
    </div>
  )
}

Pager.propTypes = {
  path: PropTypes.string.isRequired,
  currentPage: PropTypes.number.isRequired,
  numPages: PropTypes.number.isRequired,
}

export default Pager