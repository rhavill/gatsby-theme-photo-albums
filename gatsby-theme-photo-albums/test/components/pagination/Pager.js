import React from 'react'
import renderer from 'react-test-renderer'
import {cleanup, render} from '@testing-library/react'
import Pager from '../../../src/components/pagination/Pager'

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup)

describe('Pager', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Pager path='/some-path/page-2' currentPage={2} numPages={3} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('displays Previous and Next link if previous and next pages exist', () => {
    const {getByText} = render(
      <Pager path='/some-path/page-2' currentPage={2} numPages={3} />
    )
    expect(getByText('← Previous Page')).toBeTruthy()
    expect(getByText('Next Page →')).toBeTruthy()
  })
  it('only displays Next link for first page', () => {
    const {getByText, queryByText} = render(
      <Pager path='/some-path' currentPage={1} numPages={3} />
    )
    expect(queryByText('← Previous Page')).toBeNull()
    expect(getByText('Next Page →')).toBeTruthy()
  })
  it('only displays Previous link for first page', () => {
    const {getByText, queryByText} = render(
      <Pager path='/some-path/page-3' currentPage={3} numPages={3} />
    )
    expect(getByText('← Previous Page')).toBeTruthy()
    expect(queryByText('Next Page →')).toBeNull()
  })
})
