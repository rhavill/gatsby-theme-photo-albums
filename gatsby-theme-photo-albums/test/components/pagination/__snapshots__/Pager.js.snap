// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager renders correctly 1`] = `
<div
  className="css-15gdz0a-Pager"
>
  <a
    href="/some-path"
    rel="prev"
  >
    <span
      className="previous-page"
    >
      ← Previous Page
    </span>
  </a>
  <a
    href="/some-path/page-3"
    rel="next"
  >
    <span
      className="next-page"
    >
      Next Page →
    </span>
  </a>
</div>
`;
